manpages/testresolver.1
.TH TESTRESOLVER "7" "October 2012" "CompatibleOne" "Platform"
.SH NAME
testresolver \- Test the resolver
.SH SYNOPSIS
\fBtestresolver\fR
.PP
.SH DESCRIPTION
.\" Add any additional description here
.PP
This will run the CompatibleOne \fBtestresolver\fR service.
.SH AUTHOR
The CompatibleOne project.
.SH MANPAGE
Most of this documents comes from the README of ACCORDS, written by Iain James Marshall.
Written by Alexandre Lissy <alissy@mandriva.com> for the CompatibleOne project.
.SH "REPORTING BUGS"
Report CompatibleOne bugs to compatibleone@ow2.org
.br
CompatibleOne home page: <http://www.compatibleone.org>
.SH COPYRIGHT
Copyright \(co 2012 CompatibleOne.
License Apache 2.0.
.br
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.


debian/changelog
compatibleone-accords (0.20121218-1) unstable; urgency=low

  * New upstream code + more doc + sync all binary packages.

 -- Vincent Renardias <vincent@renardias.com>  Tue, 18 Dec 2012 14:43:00 +0200

compatibleone-accords (0.20121214-1) unstable; urgency=low

  * Several bug fixes regarding cosacs.

 -- Vincent Renardias <vincent@renardias.com>  Fri, 14 Dec 2012 16:58:00 +0200

compatibleone-accords (0.20121213-2) unstable; urgency=low

  * Add 'cosacs' package (client side COSACS to be run in the VM images).
  * Split 'compatibleone' in 'compatibleone-base' and 'compatibleone'.

 -- Vincent Renardias <vincent@renardias.com>  Thu, 13 Dec 2012 17:47:00 +0200

compatibleone-accords (0.20121213-1) unstable; urgency=low

  * Major renaming of packages to be fully Debian compliant and prepare the
    integration of the Cosacs client.
  * Force lib # in the dependency and bump the version to make sure we're
    not linking against a version older than today (lib ABI change).

 -- Vincent Renardias <vincent@renardias.com>  Tue, 12 Dec 2012 15:16:00 +0200

compatibleone-accords (0.20121212-1) unstable; urgency=low

  * Upstream update.
  * Major packaging update: each lib now has its own packages.

 -- Vincent Renardias <vincent@renardias.com>  Tue, 11 Dec 2012 15:47:00 +0200

compatibleone-accords (0.20121207-1) unstable; urgency=low

  * Upstream update.
  * Lots of packaging fixes (see gitorious log for the gory details)

 -- Vincent Renardias <vincent@renardias.com>  Fri, 07 Dec 2012 10:25:21 +0200

compatibleone-accords (0.20121129-1) unstable; urgency=low

  * Upstream update.

 -- Vincent Renardias <vincent@renardias.com>  Mon, 26 Nov 2012 11:43:34 +0200

compatibleone-accords (0.20121121-1) unstable; urgency=low

  * Uptream update 2012/11/21 codebase + python enabled.
  * compatibleone-example and compatibleone-testosocci are now gone.

 -- Vincent Renardias <vincent@renardias.com>  Wed, 21 Nov 2012 11:12:59 +0200

compatibleone-accords (0.20121120-1) unstable; urgency=low

  * Uptream update 2012/11/20 codebase + perm fixes.

 -- Vincent Renardias <vrenardias@citypassenger.com>  Tue, 20 Nov 2012 15:24:44 +0200

compatibleone-accords (0.20121114-2) unstable; urgency=low

  * Uptream update 2012/11/19 codebase.

 -- Vincent Renardias <vrenardias@citypassenger.com>  Mon, 19 Nov 2012 20:41:06 +0200

compatibleone-accords (0.20121114-1) unstable; urgency=low

  * Uptream update,
  * jpaasprocci -> paasprocci renaming.
  * fixes for java build on 64bit arch.

 -- Vincent Renardias <vrenardias@citypassenger.com>  Wed, 14 Nov 2012 14:24:57 +0200

compatibleone-accords (0.20121022-1) unstable; urgency=low

  * Upstream change: new accords bash script.
  * All executables now have a manpage.

 -- Vincent Renardias <vrenardias@citypassenger.com>  Mon, 22 Oct 2012 15:18:00 +0200

compatibleone-accords (0.20121015-1) unstable; urgency=low

  * Only recommend compatibleone-paprocci so everybody isn't forced to install the Java runtime.
  * psmisc dep so that we're sure killall is installed.
  * Change OpenSSL dependancy so it also builds on Debian 6.0.x
  * Change debhelper dep so the package builds on Ubuntu 9.04.
  * Commit work with Mauricio to build compatibleone-paprocci and run it with
    the appropriate autodetected java environment.
  * co-stop: can now be run from any directory.
  * Add package compatibleone-example.
  * Add compatibleone-nagios package.

 -- Vincent Renardias <vrenardias@citypassenger.com>  Mon, 15 Oct 2012 11:08:00 +0200

compatibleone-accords (0.20121008-1) unstable; urgency=low

  * More fixes/updates (see gitorious log).

 -- Vincent Renardias <vrenardias@citypassenger.com>  Mon, 8 Oct 2012 14:38:16 +0200

compatibleone-accords (0.20121004-1) unstable; urgency=low

  * More fixes/updates, Initial init script for publisher, WIP.

 -- Vincent Renardias <vrenardias@citypassenger.com>  Thu, 4 Oct 2012 11:55:16 +0200

compatibleone-accords (0.20120918-1) unstable; urgency=low

  * Various fixes/updates, WIP.

 -- Vincent Renardias <vrenardias@citypassenger.com>  Tue, 18 Sep 2012 11:03:43 +0200

compatibleone-accords (0.20120901-1) unstable; urgency=low

  * Initial Debian packaging of CompatibleOne Accords plateform.

 -- Vincent Renardias <vincent@renardias.com>  Sat, 01 Sep 2012 20:33:43 +0200


testresolver/src/testresolver.c
/* -------------------------------------------------------------------- */
/*  ACCORDS PLATFORM                                                    */
/*  (C) 2011 by Iain James Marshall (Prologue) <ijm667@hotmail.com>     */
/* -------------------------------------------------------------------- */
/* Licensed under the Apache License, Version 2.0 (the "License"); 	*/
/* you may not use this file except in compliance with the License. 	*/
/* You may obtain a copy of the License at 				*/
/*  									*/
/*  http://www.apache.org/licenses/LICENSE-2.0 				*/
/*  									*/
/* Unless required by applicable law or agreed to in writing, software 	*/
/* distributed under the License is distributed on an "AS IS" BASIS, 	*/
/* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or 	*/
/* implied. 								*/
/* See the License for the specific language governing permissions and 	*/
/* limitations under the License. 					*/
/* -------------------------------------------------------------------- */
#ifndef	_test_resolver_c
#define	_test_resolver_c

#include "occiresolver.h"

private	char *	accepts=(char *) 0;
private	int	verbose=0;
private	int	debug=0;
private	int	price=0;
private	char *	tls=(char *) 0;
private	char *	publisher=_CORDS_DEFAULT_PUBLISHER;
private	char *	agent="CO-TEST-RESOLVER";
private	char *	operator="accords";
private	char *	zone="europe";

public	int	check_debug()		{	return(debug);		}
public	int	check_verbose()		{	return(verbose);	}
public	char *	default_publisher()	{	return(publisher);	}
public	char *	default_tls()		{	return(tls);		}
public	char *	default_zone()		{	return(zone);		}
public	char *	default_operator()	{	return(operator);	}

public	int	failure( int e, char * m1, char * m2 )
{
	if ( e )
	{
		printf("\n*** failure %u",e);
		if ( m1 )
			printf(" : %s",m1);
		if ( m2 )
			printf(" : %s",m2);
		printf(" **`\n");
	}
	return( e );
}

/*	------------------------------------------------------------	*/
/*		 r e s o l v e r _ s h o w _ r e s p o n s e 		*/
/*	------------------------------------------------------------	*/
private	void	resolver_show_response( struct occi_response * rptr, char * agent )
{
	struct	occi_element * 	eptr;
	char *			urlprice;
	if ( rptr->response )
	{
		printf("   OCCI Client : %s %u %s \n",
			rptr->response->version,
			rptr->response->status,
			rptr->response->message );
	}
	for (	eptr = rptr->first;
		eptr != (struct occi_element*) 0;
		eptr = eptr->next )
	{
		if (!( eptr->name ))
			continue;
		else if (!( eptr->value ))
			continue;
		else if (!( price ))
			printf("   %s = %s \n",eptr->name,eptr->value );
		else if (!( urlprice = occi_resolve_category_price( eptr->name, eptr->value, agent, tls ) ))
			printf("   %s = %s \n",eptr->name,eptr->value );
		else
		{
			printf("   %s = %s \n\t[ %s ] \n",eptr->name,eptr->value, urlprice );
			urlprice = liberate( urlprice );
		}
	}
	return;
}



private	int	test_resolver_operation( char * host, char * agent, char * object )
{
	struct	occi_client * cptr;
	struct	occi_request * rptr;
	struct	occi_response *	aptr;

	if (!( host )) 
		return( failure(30,"please specify","--publisher name" ));

	else if (!( agent )) 
		return( failure(30,"please specify","--agent name" ));

	if ( check_verbose() )
		printf("\n   OCCI Resolver\n");

	if ( object )
	{
		if ( accepts )
		{
			occi_client_accept( accepts );
		}
		initialise_occi_resolver( host, (char *) 0, (char *) 0, (char *) 0 );
		if (( aptr = occi_resolver( object, agent )) != (struct occi_response *) 0)
		{
			resolver_show_response( aptr, agent );
			aptr=occi_remove_response( aptr );
		}
	}
	return( 0 );

}

private	int	test_resolver_command( int argc, char * argv[] )
{
	int	argi=1;
	char *	aptr;
	int	status;
	while (argi < argc)
	{
		if (!( aptr = argv[argi++] ))
			break;
		else if ( *aptr !=  '-')
		{
			if (!( status = test_resolver_operation( publisher, agent, aptr ) ))
				continue;
			else	return( status );
		}
		else 
		{
			aptr++;
			if ( *aptr ==  '-')
			{
				aptr++;
				if (!( strcmp( aptr, "verbose" ) ))
				{	verbose=1; continue;	}
				if (!( strcmp( aptr, "price" ) ))
				{	price=1; continue;	}
				else if (!( strcmp( aptr, "debug" ) ))
				{	debug=1; continue;	}
				else if (!( strcmp( aptr, "publisher" ) ))
				{
					publisher = argv[argi++];
					continue;
				}
				else if (!( strcmp( aptr, "accept" ) ))
				{
					accepts = argv[argi++];
					continue;
				}
				else if (!( strcmp( aptr, "tls" ) ))
				{
					tls = argv[argi++];
					continue;
				}
				else if (!( strcmp( aptr, "agent" ) ))
				{
					agent = argv[argi++];
					continue;
				}
				else
				{
					printf("incorrect option : %s\n",aptr);
					break;
				}
			
			}
			else
			{
				switch( *(aptr+1) )
				{
				case	'd'	:
					debug = 1;
					continue;
				case	'v'	:
					verbose = 1;
					continue;
				case	't'	:
					tls = (aptr+2);
					break;
				case	'p'	:
					publisher = (aptr+2);
					continue;
				case	'a'	:
					agent = (aptr+2);
					continue;
				default		:
					printf("incorrect option : %s \n",aptr);
					break;
				}
			}
		}
	}
	return( 0 );
}

private	int	test_resolver_banner()
{
	printf("\n   CO-TEST-RESOLVER : CompatibleOne OCCI Test Resolver : Version 1.0a.0.02");
	printf("\n   Beta Version : 04/03/2012 ");
	printf("\n   Copyright (c) 2011,2012 Iain James Marshall, Prologue \n");
	printf("\n   --verbose                         activate status  messages       ");
	printf("\n   --debug                           activate debug   messages       ");
	printf("\n   --tls       <filename>            specify TLS configuration       ");
	printf("\n   --accept    <filename>            specify ACCEPT MIME type        ");
	printf("\n   --publisher <hostname>            set publisher URL               ");
	printf("\n   --agent     <agentname>           set client agent identification \n\n");
	return(0);
}

public	int	main( int argc, char * argv[] )
{
	if ( argc == 1 )
		return( test_resolver_banner() );
	else	return( test_resolver_command(argc, argv) );
}

#endif	/* _test_resolver_c */
	/* ------------ */


